aws/resource_aws_lb_target_group_attachment.go

```diff
@@ -140,6 +140,19 @@ func resourceAwsLbAttachmentRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error reading Target Health: %s", err)
 	}
 
+	for _, targetDesc := range resp.TargetHealthDescriptions {
+		if *targetDesc.Target.Id == d.Get("target_id").(string) {
+			// These will catch targets being removed by hand (draining as we plan) or that have been removed for a while
+			// without trying to re-create ones that are just not in use. For example, a target can be `unused` if the
+			// target group isnt assigned to anything, a scenario where we don't want to continuously recreate the resource.
+			if (*targetDesc.TargetHealth.Reason == "Target.NotRegistered") || (*targetDesc.TargetHealth.Reason == "Target.DeregistrationInProgress") {
```

Comment:
Similarly here to prevent potential panics. The AWS Go SDK also provides [constants](https://docs.aws.amazon.com/sdk-for-go/api/service/elbv2/#pkg-constants) we can use instead of our own strings 👍 

```suggestion
			if targetDesc.TargetHealth == nil {
				continue
			}

			reason := aws.StringValue(targetDesc.TargetHealth.Reason)
			if reason == elbv2.TargetHealthReasonEnumTargetNotRegistered || reason == elbv2.TargetHealthReasonEnumTargetDeregistrationInProgress {
```